src/devices/SoftwareSpi/SoftwareSpi.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Device.Gpio;
+using System.Device.Spi;
+
+namespace Iot.Device.Spi
+{
+    public class SoftwareSpi : SpiDevice
+    {
+        private GpioController _controller;
+        private readonly int _clk;
+        private readonly int _miso;
+        private readonly int _mosi;
+        private readonly int _cs;
+        private readonly SpiConnectionSettings _settings;
+
+        public SoftwareSpi(int clk, int miso, int mosi, int cs, SpiConnectionSettings settings = null, GpioController controller = null)
+        {
+            _controller = controller ?? new GpioController();
+
+            _settings = settings ?? new SpiConnectionSettings(-1, -1);
+
+            _clk = clk;
+            _miso = miso;
+            _mosi = mosi;
+            _cs = cs;
```

Comment:
Can we rename this to chipSelectPin? the rest of the names might be more intuitive, but `cs` might be confusing.

Reply:
it's called like that pretty much everywhere including drawn on the board on multiple devices - I don't think this is confusing but can rename

Reply:
I guess it's fine as long as you add the xml docs since you can have more verbose info there.